### PR TITLE
Triangular bands tools

### PR DESCRIPTION
**Pull request by pabloEntropia**
This request summarizes all my changes so far plus the proposals on  on #518:
- TruangularBands: I added the new parameters to the unit test. I also tested that the power spectrum matches the magnitude bands for the unit spectrum case.
- MelBands: I left the regression test as it was to test that if you don't specify any parameter, the default configuration matches the former behavior. I included another regression test, using an HTK output provided by @georgid to test the "htk styled" Mel bands 

Things to do:
- I've seen that MFCC test has one error (because of the last TriangularBands exception added) and one failure, however as the MelBands  is passing the old  regression test is  with the old parameters my changes shouldn't have affected it however I will try to understand why and repair it.
- I've to fix the normalization of DCT type III in order to behave like Librosa

### CODE REVIEW

**Comment by dbogdanov** [comment]
The MFCC regression test was failing for some time already. Create a new regression test for MFCC from scratch.

**Comment by pabloEntropia** [comment]
I have fixed the errors on  the MFCC unit test and created a new regression test  for the new HTK style implementation comparing with the MFCC calculated by Librosa. Should I remove the old regression test?

File: test/src/unittest/spectral/test_mfcc.py [hunk 1]
```diff
@@ -38,23 +39,58 @@ def testRegression(self):
...
+            bands, mfcc = self.InitMFCC(size)(ones(1025))
             self.assertEqual(len(mfcc), size )
             self.assertEqual(len(bands), 40 )
             self.assert_(not any(numpy.isnan(mfcc)))
             self.assert_(not any(numpy.isinf(mfcc)))
             size -= 1
 
+    def testRegressionHtkMode(self):
+        audio = essentia.array(np.loadtxt(join(filedir(), 'mfcc', 'audio.txt')))
+        trueMFCC = np.loadtxt(join(filedir(), 'mfcc', 'true_mfcc.txt'))
```
Comment (dbogdanov) [inline test/src/unittest/spectral/test_mfcc.py]: Better file name: "true_mfcc_htk.txt"?

**Comment by dbogdanov** [comment]
The ```test/src/unittest/spectral/mfcc/audio.txt``` and ```true_mfcc``` files are missing.

**Comment by dbogdanov** [comment]
Please add these files before I do the merge.

**Comment by dbogdanov** [comment]
We'll have to fix regression test for htk MFCC. New test data is too large (33M for encoding 30 seconds of audio stored in plain text  + 1.5M for reference MFCC values in frames). This is not a good practice to store that in our github repository if we can avoid that.

Do this instead:
- [remove the last commit from git history in local and remote](http://stackoverflow.com/a/8225166/603642) (ef5e616) (be careful!)
- do a new commit with updated test
    - use a test audio file from ```test/audio/recorded/``` with MonoLoader
    - pre-compute HTK reference values, however, do not store frames, but only the mean. These would be only 20 values, you can store them in the code itself.

**Pull request merged**